src/factorizations/arnoldi.jl

```diff
@@ -142,7 +142,7 @@ function initialize(iter::ArnoldiIterator; verbosity::Int=0)
     α = inner(x₀, Ax₀) / (β₀ * β₀)
     T = typeof(α)
     # this line determines the vector type that we will henceforth use
-    v = add!!(zerovector(Ax₀, T), x₀, 1 / β₀)
+    v = add!!(scale(Ax₀, zero(α)), x₀, 1 / β₀)
```

Comment:
Is this because `inner` can now return a number which is not the `scalartype` of the vectors? It might be convenient to mention that here, because next time I look at this I will definitely not remember.

Perhaps we should also at some point centralize that logic somewhere, as this is repeated in many places in the code and is a bit messy anyways

Reply:
Yes, `scalartype` used to mean basically the scalar field over which the vector space was defined, and the whole real business makes this ambiguous. So indeed, if `T` is simply determined from the inner product of vectors, it will be real for `RealVec`, but the scalartype is actually still complex. Given that `scalartype` is now part of the required interface (even though I don't think we use it), we could also do something like `T = promote(typeof(α), scalartype(Ax₀))`. Although then we have to be careful furtheron, where `T` is also used to initialise the matrix `H` and the other data related to the Krylov subspace, which should be fully determined by the return type of the inner product alone. 

Actually, it seems the GMRES implementation was already doing something similar in its initialisation stage, so that one did not have an issue.

Reply:
I've added some comments, so I think this is ok now?